crates/services/executor/src/executor.rs

```diff
@@ -724,13 +728,16 @@ where
             return Err(ExecutorError::DaHeightExceededItsLimit)
         };
 
+        let mut root_calculator = MerkleRootCalculator::new();
+
         for da_height in next_unprocessed_da_height..=header.da_height.0 {
             let da_height = da_height.into();
             let events = self
                 .relayer
                 .get_events(&da_height)
                 .map_err(|err| ExecutorError::RelayerError(err.into()))?;
             for event in events {
+                root_calculator.push(event.hash().as_ref());
```

Comment:
Hmm, I'm curious: does Ethereal RPC guarantee that the order of events is the same as they were emitted? @MitchTurner , Could you double-check that?=)

Reply:
Looking at ethers, there is a field for indexing fields within each block. We could sort all the logs by this in the relayer to play it safe.

![image](https://github.com/FuelLabs/fuel-core/assets/794823/90e6d019-9adb-479e-8eb5-f65452679c54)

Reply:
We could also consider lexicographically sorting all the event hashes before merklizing

Reply:
The order we receive the events shouldn't be dependent on the Ethereal RPC specifically, rather our implementation of `Relayer` as a whole. 

IF the order isn't deterministic, then I agree with @Voxelot that we could just do some deterministic sorting in our `Relayer` adapter.

Do we want to do that as part of this PR?

Reply:
Or we could just order them after we get them from the `Relayer`.

Reply:
The best ordering Imo is to use log_index, as it ensures fair ordering of event processing. I'd prefer to avoid muddling our events in the executor with ethereum specific fields, so I'd be fine with making a separate PR to ensure ordering on the relayer side.

The alternative is to add some kind of ordering field to events, and making the executor sort them every time. However, this would also incur additional consensus overhead which I think we could avoid.